File: api/src/main/java/com/ruby/rubia_server/core/dto/ConversationSummaryDTO.java
package com.ruby.rubia_server.core.dto;

import com.ruby.rubia_server.core.enums.Channel;
import com.ruby.rubia_server.core.enums.ConversationStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ConversationSummaryDTO {
    
    private UUID id;
    private String customerName;
    private String customerPhone;
    private String assignedUserName;
    private ConversationStatus status;
    private Channel channel;
    private Integer priority;
    private LocalDateTime updatedAt;
    private Long unreadCount;
    private String lastMessageContent;
    private LocalDateTime lastMessageTime;
}

File: api-dotnet/Rubia.Server/DTOs/ConversationSummaryDto.cs
using Rubia.Server.Enums;

namespace Rubia.Server.DTOs;

public class ConversationSummaryDto
{
    public Guid Id { get; set; }
    public string? CustomerName { get; set; }
    public string? CustomerPhone { get; set; }
    public string? AssignedUserName { get; set; }
    public ConversationStatus Status { get; set; }
    public Channel Channel { get; set; }
    public int? Priority { get; set; }
    public DateTime UpdatedAt { get; set; }
    public long UnreadCount { get; set; }
    public string? LastMessageContent { get; set; }
    public DateTime? LastMessageTime { get; set; }
    public string? ChatLid { get; set; }
}